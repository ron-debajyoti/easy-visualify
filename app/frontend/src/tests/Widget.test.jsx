import React from 'react';
import renderer from 'react-test-renderer';
import { unmountComponentAtNode } from 'react-dom';
import PlayWidget from '../components/Widget';
import { fakePlaylistUri, fakeTrackUri } from './fakedata';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders tracks without crashing', () => {
  const div = document.createElement('div');
  const type = 'song';
  const uri = fakeTrackUri;
  const title = 'Test 1';
  const tree = renderer.create(<PlayWidget type={type} uri={uri} title={title} />, div).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders playlists without crashing', () => {
  const div = document.createElement('div');
  const type = 'playlist';
  const uri = fakePlaylistUri;
  const title = 'Test 2';
  const tree = renderer.create(<PlayWidget type={type} uri={uri} title={title} />, div).toJSON();
  expect(tree).toMatchSnapshot();
});
