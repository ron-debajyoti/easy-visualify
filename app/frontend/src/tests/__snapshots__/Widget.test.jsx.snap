// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders playlists without crashing 1`] = `
.c0 {
  min-width: 35vh;
  min-height: 100vh;
}

<iframe
  className="c0 widget-playlist"
  src="https://open.spotify.com/embed/playlist/spotify:playlist:6sDvv5SNASdsfeeww"
  title="Test 2"
/>
`;

exports[`renders tracks without crashing 1`] = `
@media (max-width:768px) {
  .c0 {
    width: 250px;
    height: 70px;
  }
}

@media (min-width:768px) {
  .c0 {
    width: 250px;
    height: 70px;
  }
}

<embed
  className="c0 widget-song"
  src="https://open.spotify.com/embed?uri=spotify:track:4KD0lLJ4OGonZhBeKtct9I&theme=white"
  title="Test 1"
/>
`;
